TrafficSimulator/Assets/Utilities/DayNightManager.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Simulation
+{
+    public enum SunMode
+    {
+        Time,
+        Day,
+        Night
+    }
+
+    public class DayNightManager : MonoBehaviour
+    {
+        public SunMode _sunMode = SunMode.Time;
+
+        void Start()
+        {
+            SetSunLocation(_sunMode);
+
+            // Subscribe to time manager events for hour
+            TimeManager.OnHourChanged += MoveSunByTime;
+        }
+
+        private void SetSunLocation(SunMode mode)
+        {
+            switch (mode)
+            {
+                case SunMode.Time:
```

Comment:
This needs to set the rotation according to the current time. At the moment the sun will not be updated even if `sunMode` is Time, it will stay whatever the default position is until an hour passes until it corrects itself.

So in the worst case the sun will be in the incorrect position for an entire hour in play mode until it gets fixed